### PR TITLE
[flang][OpenMP] Move clause/object conversion to happen early, in genOMP

### PR DESCRIPTION
**Pull request by kparzysz**
This removes the last use of genOmpObjectList2, which has now been removed.

### CODE REVIEW

**Review by luporl — APPROVED** [review]
LGTM

File: flang/lib/Lower/OpenMP/OpenMP.cpp [hunk 1]
```diff
@@ -2357,44 +2332,44 @@ genOMP(Fortran::lower::AbstractConverter &converter,
...
-        !std::get_if<Fortran::parser::OmpClause::Copyin>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::Shared>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::Threads>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::Map>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::UseDevicePtr>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::UseDeviceAddr>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::ThreadLimit>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::NumTeams>(&clause.u) &&
-        !std::get_if<Fortran::parser::OmpClause::Simd>(&clause.u)) {
+    if (!std::get_if<clause::If>(&clause.u) &&
```
Comment (skatrak) [inline flang/lib/Lower/OpenMP/OpenMP.cpp:2345]: Nit: I'd suggest taking the opportunity of already updating all these lines to sort them alphabetically and possibly replace `std::get_if` with `std::holds_alternative`, but feel free to ignore (same thing below for `endClauses`).

**Review by skatrak — APPROVED** [review]
Thank you Krzysztof, LGTM. I have a couple of suggestions, but this should be ready if you don't agree with them.

**Comment by kparzysz** [comment]
We can switch to ArrayRef in another PR.

File: flang/lib/Lower/OpenMP/OpenMP.cpp (hunk 1 above)
Comment (kparzysz) [reply flang/lib/Lower/OpenMP/OpenMP.cpp:2185]: I did that in https://github.com/llvm/llvm-project/pull/89090.

**Pull request merged**